app.py

```diff
@@ -1,72 +1,179 @@
-            print ('end ...')
-        except TypeError:
-            print ('Not a valid room type try "living or office".')
-        except:
-            print('Not a valid usage')
+        """Usage: create_room <room_type> <room_names>...
+        """
+        room_purpose = arg.get('<room_type>')
+        room_names = arg.get('<room_names>')
+        Dojo().create_room(room_purpose, room_names)
 
     @docopt_cmd
     def do_add_person(self, arg):
-        """Usage: add_person (<fname> <lname> <FELLOW/STAFF> [<wants_accommodation>])
+        """Usage: add_person <fname> <lname> <FELLOW/STAFF> [<wants_accommodation>]
         """
-        try:
-            if(arg.get('<wants_accommodation>')):
-                Dojo().add_person(arg.get('<fname>'), arg.get('<lname>'), arg.get('<FELLOW/STAFF>'), arg.get('<wants_accommodation>','n')  )
-                print('end valid wants_accommodation...')
+        fname = arg.get('<fname>')
+        lname = arg.get('<lname>')
+        person_type = arg.get('<FELLOW/STAFF>')
+        wants_accommodation = arg.get('<wants_accommodation>')
+        if fname.isalpha() and lname.isalpha():
+            if person_type == 'staff':
+                Dojo().add_person(fname, lname, 'staff')
+            elif person_type == 'fellow':
+                if wants_accommodation not in ('y','Y'):
+                    Dojo().add_person(fname, lname, 'fellow')
+                else:
+                    Dojo().add_person(fname, lname, 'fellow', 'y')
             else:
-                print (arg)
-                Dojo().add_person(arg.get('<fname>'), arg.get('<lname>'), arg.get('<FELLOW/STAFF>'), 'n')
-                print('end None for want_accommodation...')
-        except TypeError:
-            print ('<FELLOW/STAFF> type person can only be a Fellow or Staff')
-        except:
-            print('Not a valid usage')
+                print('Invalid!!! Dojo Occupants are either Fellows or Staff')
+        else:
+            print('Names can only be string character')
+
+        # Dojo().add_person()
+    @docopt_cmd
+    def do_print_room(self, arg):
+        """Usage: print_room <room_name>
+        """
+        room_name = arg.get('<room_name>')
+        Dojo().print_room(room_name)
+
+    @docopt_cmd
+    def do_print_allocations(self, arg):
+        """Usage: print_allocations [(-o <filename>)]
+        """
+        filename = arg.get('<filename>')
+        if not (filename):
+            Dojo().print_allocations()
+        else:
+            Dojo().print_allocations(filename)
+
+    @docopt_cmd
+    def do_print_unallocated(self, arg):
+        """Usage: print_unallocated [(-o <filename>)]
+        """
+        filename = arg.get('<filename>')
+        if not (filename):
+            Dojo().print_unallocated()
+        else:
+            Dojo().print_unallocated(filename)
+
+    @docopt_cmd
+    def do_people_id(self, arg):
+        """Usage: people_id
+        """
+        Dojo().people_id()
+        pass
```

Comment:
@tsolarin Okay I have taken note of all your remarks, thanks for taking out the time too.